### ISSUE TITLE
what's difference between authorized_keys and authorized_keys

### ISSUE THREAD

**Issue by shanjunmei** [NHAS/reverse_ssh#55]
as REAME flow both of that two file will be same content ,but when i try start the server ,told me `                                ssh-ed25519 �Y���K��+��bwB ���\�A� is present in both authorized_controllee_keys and authorized_keys. It should only be in one.
`
what's wrong

**Comment by NHAS** [comment]
Hi,
The readme states that `authorized_keys` requires your `ssh` public key, the key that the `ssh` offer when connecting to the server. This is the same as regular `ssh`, detailed here: http://man.he.net/man5/authorized_keys

`authorized_controllee_keys` contains the public key of the client binary, this is generated when building with `make`. 

If a public key is in both the `authorized_keys` and `authorized_controllee_keys` files, it would mean that a client could control the server, or that an administrator may be a client binary. 
This makes it impossible for the server to determine whether the management interface should be provided to said user.


TL;DR

Put your key in `authorized_keys`, put the client key in `authorized_controllee_keys`, both files are not the same.

**Comment by shanjunmei** [comment]
thanks~

**Comment by shanjunmei** [comment]
> Hi, The readme states that `authorized_keys` requires your `ssh` public key, the key that the `ssh` offer when connecting to the server. This is the same as regular `ssh`, detailed here: http://man.he.net/man5/authorized_keys
> 
> `authorized_controllee_keys` contains the public key of the client binary, this is generated when building with `make`.
> 
> If a public key is in both the `authorized_keys` and `authorized_controllee_keys` files, it would mean that a client could control the server, or that an administrator may be a client binary. This makes it impossible for the server to determine whether the management interface should be provided to said user.
> 
> TL;DR
> 
> Put your key in `authorized_keys`, put the client key in `authorized_controllee_keys`, both files are not the same.

hi DR，i'm sorry bother you again ,but when i use current 1.05 version  face a issue ,when i try  access the server port: 3232 ,i get a error ` Permission denied (publickey).` .before i start the server i already put publickey in file `authorized_keys`

**Comment by NHAS** [comment]
Cant replicate.

**Comment by shanjunmei** [comment]
> Cant replicate.

this is my operate steps
1. git clone https://github.com/NHAS/reverse_ssh.git
2. cd reverse_ssh
3. make 
4. ssh-keygen 
5. cp ~/.ssh/id_rsa.pub(previous step generate result) bin/authorized_controllee_keys
6. cd bin
7. ./server --insecure 0.0.0.0:3232
8.  open another terminal in same computer(ubuntu based)
9. ssh 127.0.0.1 -p 3232
10. get confirm information and type yes ,got this message ,`Warning: Permanently added '[127.0.0.1]:3232' (ED25519) to the list of known`
11. get `Permission denied (publickey)` error hosts. )
12. Is there something wrong with my steps?

**Comment by NHAS** [comment]
Actually I take that back, I can replicate this. 
However, it seems that this is an upstream failure, https://github.com/golang/go/issues/49952

The golang crypto/ssh library does not support the new rsa-sha256 algorithms required by openssh. 

To fix this, you will need to generate a different key. I would recommend the following: 
```
ssh-keygen -t ed25519
```

**Comment by NHAS** [comment]
I'll add this to the readme as its quite frustrating

**Comment by NHAS** [comment]
https://github.com/NHAS/reverse_ssh#permission-denied-publickey

**Comment by shanjunmei** [comment]
> Actually I take that back, I can replicate this. However, it seems that this is an upstream failure, [golang/go#49952](https://github.com/golang/go/issues/49952)
> 
> The golang crypto/ssh library does not support the new rsa-sha256 algorithms required by openssh.
> 
> To fix this, you will need to generate a different key. I would recommend the following:
> 
> ```
> ssh-keygen -t ed25519
> ```

thanks you so much ,for this issue, is there any log in other places that can reflect this? I still can't find this information when I run it in debug mode.

**Comment by NHAS** [comment]
Im not sure what you mean by this. 

Unfortunately the logging for the rssh server cant capture this, as it happens within the library itself. 

The ssh tool however does tell us that there is no matching signature algorithm

**Comment by NHAS** [comment]
I'm not entirely sure what you mean by any log for this.

The ssh tool tells us that there is no supported signature algorithm.

On the RSSH server side of things, I can't log this as it's handled in the
ssh library itself. Which is a bit stink
